### PR TITLE
Open file in UTF-8 encoding (#6919)

### PR DESCRIPTION
**Pull request by flyspirit99**
FileCallbackHandler cannot handle some language, for example: Chinese. 
Open file using UTF-8 encoding can fix it.
@agola11
  
**Issue**: #6919 
**Dependencies**: NO dependencies,

### CODE REVIEW

File: libs/langchain/langchain/callbacks/file.py [hunk 1]
```diff
@@ -13,7 +13,7 @@ def __init__(
         self, filename: str, mode: str = "a", color: Optional[str] = None
     ) -> None:
         """Initialize callback handler."""
-        self.file = cast(TextIO, open(filename, mode))
+        self.file = cast(TextIO, open(filename, mode, encoding = "utf-8"))
```
Comment (eyurtsev) [inline libs/langchain/langchain/callbacks/file.py:16]: I don't think this has an effect in python 3, as of python 3 open applies a default utf-8 decoding to content

**Review by eyurtsev — CHANGES_REQUESTED** [review]

**Comment by eyurtsev** [comment]
Thanks for the contribution! It might be helpful if you share details about the error you're seeing and/or include a unit test illustrating that handles both the happy and the bad paths

**Comment by flyspirit99** [comment]
> Thanks for the contribution! It might be helpful if you share details about the error you're seeing and/or include a unit test illustrating that handles both the happy and the bad paths

I used custom prompt, put some Chinese in it.  Then I created a chain using this prompt, with verbose set to True. In the log file, the Chinese characters in the prompt will display corrupted text. Please see the screenshot below.

image

By specifying encoding="utf-8" when open file, the issue was fixed.  

image

File: libs/langchain/langchain/callbacks/file.py (hunk 1 above)
Comment (flyspirit99) [reply libs/langchain/langchain/callbacks/file.py:16]: It depends on the OS environment.  Please checkout this link https://docs.python.org/3/library/functions.html#open
and this part specifically

> The default encoding is platform dependent (whatever [locale.getencoding()](https://docs.python.org/3/library/locale.html#locale.getencoding) returns), 

There is a plan to use UTF-8 as default encoding for all OS. But we have to wait until 3.15 release as metioned in PEP 686
https://peps.python.org/pep-0686/#motivation

Comment (eyurtsev) [reply libs/langchain/langchain/callbacks/file.py:16]: Got it! Thanks didn't realize this!

**Review by eyurtsev — APPROVED** [review]

**Comment by baskaryan** [comment]
thanks @flyspirit99!

**Comment by flyspirit99** [comment]
Thanks. @baskaryan  
There was a format issue causing CI/CD fails. I will run "make lint" beforehand next time. Sorry for the hassle.

**Pull request merged**